test/functional/sqlsrv/sqlsrv_batch_query.phpt

```diff
@@ -0,0 +1,179 @@
+--TEST--
+Test a batch query with different cursor types
+--DESCRIPTION--
+Verifies that batch queries don't work with dynamic, static, and keyset
+server-side cursors, and checks that correct column and row counts are
+returned otherwise
+--SKIPIF--
+<?php require('skipif.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon.inc');
+
+$cursors = array('forward', 'dynamic', 'static', 'keyset', 'buffered');
+
+$expectedCols = 1;
+$expectedRows = 5;
+$expectedResultSets = 5;
+
+//expected errors
+$noCursor = array(array('01000', 16954, '[Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Executing SQL directly; no cursor.'),
+                  array('01S02', 0, '[Microsoft][ODBC Driver 17 for SQL Server]Cursor type changed'));
+$wrongCursor = array(array('IMSSP', -50, 'This function only works with statements that have static or keyset scrollable cursors.'));
+$noNextResult = array(array('IMSSP', -22, 'There are no more rows in the active result set.  Since this result set is not scrollable, no more data may be retrieved.'));
+
```

Comment:
Please omit `ODBC Driver 17 for SQL Server` in the expected error messages because this test should pass when testing against other ODBC versions. Thus, comparing a substring of an error message should suffice (such as using fnmatch(), etc.)